devicetypes/drzwave/ezmultipli.src/ezmultipli.groovy

```diff
@@ -247,15 +248,22 @@ def zwaveEvent(switchcolorv3.SwitchColorReport cmd) {
 		result << createEvent(name: "color", value: hexColor)
 		// Send the color as hue and saturation
 		def hsv = rgbToHSV(*colors)
-		result << createEvent(name: "hue", value: hsv.hue)
-		result << createEvent(name: "saturation", value: hsv.saturation)
-		// Reset the values
-		RGB_NAMES.collect { state.colorReceived[it] = null}
+		if (state.setColor.red == state.colorReceived.red && state.setColor.green == state.colorReceived.green && state.setColor.blue == state.colorReceived.blue) {
+			unschedule()
+			result << createEvent(name: "hue", value: hsv.hue)
+			result << createEvent(name: "saturation", value: hsv.saturation)
+		} else {
+			runIn(2, "sendColorQueryCommands", [overwrite: true])
```

Comment:
I can see this easily ending up in an endless loop. For instance if the device never actually received our setColor command.

It *might* be better to attempt to send our set command again instead of querying the colors?

Thinking back about the original issue. Perhaps the problem is with the logic behind the nulling out of `state.colorReceived` in `parse()`?

Reply:
You mean here https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/2d1c56701c1aebb70c2e04df30f633cf6ac01f3b/devicetypes/drzwave/ezmultipli.src/ezmultipli.groovy#L159 ? I doubt that's the case here, it looks like it's only a backup map initializer if one in _installed()_ failed some reason.